lib/temporal/workflow/executor.rb

```diff
@@ -36,6 +36,10 @@ def run
         return state_manager.commands
       end
 
+      def process_query(query, args)
```

Comment:
See above. Looks like dispatcher is not accessible from within task processor, so do the special query handling here.